lua/lspconfig/jdtls.lua

```diff
@@ -1,61 +1,53 @@
 local configs = require 'lspconfig/configs'
 local util = require 'lspconfig/util'
 local handlers = require 'vim.lsp.handlers'
-local path = util.path
 
 local server_name = 'jdtls'
 
-local cmd = {
-  util.path.join(tostring(vim.fn.getenv 'JAVA_HOME'), '/bin/java'),
-  '-Declipse.application=org.eclipse.jdt.ls.core.id1',
-  '-Dosgi.bundles.defaultStartLevel=4',
-  '-Declipse.product=org.eclipse.jdt.ls.core.product',
-  '-Dlog.protocol=true',
-  '-Dlog.level=ALL',
-  '-Xms1g',
-  '-Xmx2G',
-  '-jar',
-  tostring(vim.fn.getenv 'JAR'),
-  '-configuration',
-  tostring(vim.fn.getenv 'JDTLS_CONFIG'),
-  '-data',
-  tostring(vim.fn.getenv 'WORKSPACE'),
-  '--add-modules=ALL-SYSTEM',
-  '--add-opens java.base/java.util=ALL-UNNAMED',
-  '--add-opens java.base/java.lang=ALL-UNNAMED',
+local sysname = vim.loop.os_uname().sysname
+local env = {
+  JAVA_HOME = os.getenv 'JAVA_HOME',
+  WORKSPACE = os.getenv 'WORKSPACE',
+  HOME = os.getenv 'HOME',
+  JDTLS_HOME = os.getenv 'JDTLS_HOME',
 }
 
---- The presence of one of these files indicates a root directory.
---
--- We search for configuration files of the most common Java build systems. We
--- could add more build systems, such as Make, but let's keep things simple for
--- now.
-local root_files = {
-  -- Single-module projects
-  {
-    'build.xml', -- Ant
-    'pom.xml', -- Maven
-    'settings.gradle', -- Gradle
-    'settings.gradle.kts', -- Gradle
-  },
-  -- Multi-module projects
-  { 'build.gradle', 'build.gradle.kts' },
-}
+local function get_java_executable()
+  local executable = env.JAVA_HOME and env.JAVA_HOME .. '/bin/java' or 'java'
+
+  return sysname:match 'Windows' and executable .. '.exe' or executable
```

Comment:
Will the path concattenation above actually work under windows? Before this used `util.path.join` 
(Same also happens in other places)

Reply:
I'm currently doing some tests in Windows, will confirm once it works